### DISCUSSION TITLE
Read and Write Handlers in Device Service SDK

### DISCUSSION THREAD

**Discussion by sounak-1989** [edgexfoundry/community#8] [Q&A]
Hi guys,

I have multiple profiles in my Device Service. The profiles are chosen based on what kind of devices are discovered connected to an SPI based daisy chain network.

I recently figured out that the only way to assign profile to a newly discovered device is assigning identifiers in the provision watcher which in turn filters based on the profile mentioned in the provision watcher.

Now the question is if the different profiles have different resources, when writing my read and write handlers what's the ideal way to segregate the requests based on profile. I see that there is a protocol argument that gets passed in to the handlers. I am assuming that is a way. But let me know if it is something else.

Thanks

**Comment by iain-anderson** [comment]
Hi, the `protocol` argument should hold the elements needed to access the device - these are defined in the ProtocolProperties part of a device definition. The `Attributes` value in the `reqs` argument should define how to access an individual resource within a device. These are the attributes specified in the device profile.
